metaseq/distributed/utils.py

```diff
@@ -63,6 +63,7 @@ def _infer_torch_distributed_launch_init(cfg: DistributedTrainingConfig):
     cfg.distributed_init_method = "env://"
     cfg.distributed_world_size = int(os.environ["WORLD_SIZE"])
     cfg.distributed_rank = int(os.environ["RANK"])
+    cfg.device_id = int(os.environ.get("LOCAL_RANK", 0))
```

Comment:
Why the default 0? It can silently do the wrong thing. Better to not have default so user knows this var needs to be set?

Reply:
`LOCAL_RANK` is not among the required variables for torch distributed (see [this documentation](https://pytorch.org/docs/stable/distributed.html#environment-variable-initialization)), and it's only available with Torch Elastic (i.e. `torchrun`), the recommended launcher for distributed jobs. I was just trying not to fail single-GPU jobs here with the default value 0, but your suggestion makes senses as we already cover it [here](https://github.com/facebookresearch/metaseq/blob/main/metaseq/distributed/utils.py#L53).